tests/platform_tests/link_flap/test_cont_link_flap.py

```diff
@@ -150,32 +164,41 @@ def test_cont_link_flap(self, request, duthosts, nbrhosts, enum_rand_one_per_hws
 
         # Check the FRR daemons memory usage at end
         end_time_frr_daemon_memory = {}
-        incr_frr_daemon_memory_threshold = {}
+        incr_frr_daemon_memory_threshold = defaultdict(lambda: [])
+
         for daemon in frr_demons_to_check:
-            incr_frr_daemon_memory_threshold[daemon] = 10 if tbinfo["topo"]["type"] in ["m0", "mx"] else 5
-            min_threshold_percent = 1 / float(start_time_frr_daemon_memory[daemon]) * 100
-            if min_threshold_percent > incr_frr_daemon_memory_threshold[daemon]:
-                incr_frr_daemon_memory_threshold[daemon] = math.ceil(min_threshold_percent)
-            logging.info(f"The memory increment threshold for frr daemon {daemon} "
-                         f"is {incr_frr_daemon_memory_threshold[daemon]}%")
+            for asic_index, asic_frr_memory in enumerate(start_time_frr_daemon_memory[daemon]):
+                incr_frr_daemon_memory_threshold[daemon].append(10 if tbinfo["topo"]["type"] in ["m0", "mx"] else 5)
+
+                min_threshold_percent = 1 / float(start_time_frr_daemon_memory[daemon][asic_index]) * 100
+
+                if min_threshold_percent > incr_frr_daemon_memory_threshold[daemon][asic_index]:
+                    incr_frr_daemon_memory_threshold[daemon][asic_index] = math.ceil(min_threshold_percent)
+
+                logging.info(f"The memory increment threshold for frr daemon {daemon}-asic{asic_index} "
+                             f"is {incr_frr_daemon_memory_threshold[daemon][asic_index]}%")
+
         for daemon in frr_demons_to_check:
             # Record FRR daemon memory status at end
             end_time_frr_daemon_memory[daemon] = self.get_frr_daemon_memory_usage(duthost, daemon)
             logging.info(f"{daemon} memory usage at end: \n%s", end_time_frr_daemon_memory[daemon])
 
             # Calculate diff in FRR daemon memory
-            incr_frr_daemon_memory = \
-                float(end_time_frr_daemon_memory[daemon]) - float(start_time_frr_daemon_memory[daemon])
-            logging.info(f"{daemon} absolute difference: %d", incr_frr_daemon_memory)
-
-            # Check FRR daemon memory only if it is increased else default to pass
-            if incr_frr_daemon_memory > 0:
-                percent_incr_frr_daemon_memory = \
-                    (incr_frr_daemon_memory / float(start_time_frr_daemon_memory[daemon])) * 100
-                logging.info(f"{daemon} memory percentage increase: %d", percent_incr_frr_daemon_memory)
-                pytest_assert(percent_incr_frr_daemon_memory < incr_frr_daemon_memory_threshold[daemon],
-                              f"{daemon} memory increase more than expected: "
-                              f"{incr_frr_daemon_memory_threshold[daemon]}%")
+            for asic_index, asic in enumerate(duthost.asics):
```

Comment:
Can we enumerate `end_time_frr_daemon_memory` instead of `duthost.asics` to keep it consistent please?